dataproxy/service.go

```diff
@@ -86,6 +86,41 @@ func (s Service) CreateUploadLocation(ctx context.Context, req *service.CreateUp
 	}, nil
 }
 
+// CreateDownloadLocation creates a temporary signed url to allow callers to download content.
+func (s Service) CreateDownloadLocation(ctx context.Context, req *service.CreateDownloadLocationRequest) (
+	*service.CreateDownloadLocationResponse, error) {
+
+	if expiresIn := req.ExpiresIn; expiresIn != nil {
+		if !expiresIn.IsValid() {
+			return nil, fmt.Errorf("expiresIn [%v] is invalid", expiresIn)
+		}
+
+		if expiresIn.AsDuration() < 0 {
+			return nil, fmt.Errorf("expiresIn [%v] should not less than 0",
+				expiresIn.AsDuration().String())
+		} else if expiresIn.AsDuration() > s.cfg.Download.MaxExpiresIn.Duration {
+			return nil, fmt.Errorf("expiresIn [%v] cannot exceed max allowed expiration [%v]",
+				expiresIn.AsDuration().String(), s.cfg.Download.MaxExpiresIn.String())
+		}
+	} else {
+		req.ExpiresIn = durationpb.New(s.cfg.Download.MaxExpiresIn.Duration)
+	}
+
+	resp, err := s.dataStore.CreateSignedURL(ctx, storage.DataReference(req.NativeUrl), storage.SignedURLProperties{
+		Scope:     stow.ClientMethodGet,
+		ExpiresIn: req.ExpiresIn.AsDuration(),
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a signed url. Error: %w", err)
+	}
+
+	return &service.CreateDownloadLocationResponse{
+		SignedUrl: resp.URL.String(),
+		ExpiresAt: timestamppb.New(time.Now().Add(req.ExpiresIn.AsDuration())),
```

Comment:
Is this valid  that we can accurately say here . The library sets this time for signed url if its google 
https://github.com/flyteorg/stow/blob/9c3f5f9ea24966206f0b2edc608da6ca803420e3/google/container.go#L55

Reply:
it's not accurate, but I think it's okay for now because we can't get the expiry time from [SignedURLResponse](https://github.com/flyteorg/flytestdlib/blob/d0a5767d30272c065a075a283fd2ba563834baca/storage/storage.go#L56-L58)
I can add it in a separate PR.